### ISSUE TITLE
Placeholder names on NumberBox result in strange narrator behavior

### ISSUE THREAD

**Issue by chingucoding** [microsoft/microsoft-ui-xaml#3470]
**Describe the bug**

When the value of the NumberBoxes is set, narrator still reads out the placeholder, resulting in announcment likes "0.00, edit 100".

@SavoySchuler @YuliKl FYI as I am not sure if this is the best behavior for NumberBox.

**Steps to reproduce the bug**

1. Open NumberBox page
2. Start narrator
3. Enter value in NumberBox
4. Switch to and from that NumberBox

**Expected behavior**

Narrator should read out something less confusing
**Actual behavior**

Narrator announces somewhat confusing content.
**Screenshots**

**Version Info**

Gallery version (found on Settings page): 

Latest debug build

Windows 10 version:
- Insider Build (20236)

Device form factor:
- Desktop

**Additional context**

**Comment by ranjeshj** [comment]
I'm guessing the narrator behavior is coming from TextBox since we just template bind placeholder text, but I'm not seeing the same behavior in TextBox. It is possible that this is caused by the way TextBox is being used by NumberBox.

**Comment by chingucoding** [comment]
Yes, this seems to be an issue with the TextBox jumping in and providing the UIA name here instead of the NumberBox, which in it's current implementation wouldn't even report the value. Is it intended that the NumberBoxAutomationPeer does not read out it's current value if Narrator would use it?
